backend/src/nodes/nodes/image_channel/palette_extraction.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from enum import Enum
+
+import numpy as np
+
+from . import category as ImageChannelCategory
+from ...impl.dithering.palette import (
+    distinct_colors_palette,
+    kmeans_palette,
+    median_cut_palette,
+)
+from ...node_base import NodeBase, group
+from ...node_factory import NodeFactory
+from ...properties import expression
+from ...properties.inputs import ImageInput, EnumInput, NumberInput
+from ...properties.outputs import ImageOutput
+
+
+class PaletteExtractionMethod(Enum):
+    ALL = "all"
+    KMEANS = "k-means"
+    MEDIAN_CUT = "median"
+
+
+PALETTE_EXTRACTION_METHOD_LABELS = {
+    PaletteExtractionMethod.ALL: "All distinct colors",
+    PaletteExtractionMethod.KMEANS: "K-Means",
+    PaletteExtractionMethod.MEDIAN_CUT: "Median cut",
+}
+
+
+@NodeFactory.register("chainner:image:palette_from_image")
+class PaletteFromImage(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = (
+            "Use an image to create a palette.  A palette is an image with one row."
+        )
+        self.inputs = [
+            ImageInput(),
+            EnumInput(
+                PaletteExtractionMethod,
+                option_labels=PALETTE_EXTRACTION_METHOD_LABELS,
+                default_value=PaletteExtractionMethod.KMEANS,
+            ).with_id(1),
+            group(
+                "conditional-enum",
+                {
+                    "enum": 1,
+                    "conditions": [
+                        [
+                            PaletteExtractionMethod.KMEANS.value,
+                            PaletteExtractionMethod.MEDIAN_CUT.value,
+                        ]
+                    ],
+                },
+            )(
+                NumberInput(
+                    "Palette Size",
+                    minimum=2,
+                    default=8,
+                ).with_id(2),
+            ),
+        ]
+        self.outputs = [
+            ImageOutput(
```

Comment:
Can you give this output a label of "Palette"?

Reply:
Of course!